db/system_distributed_keyspace.cc

```diff
@@ -160,6 +161,33 @@ system_distributed_keyspace::system_distributed_keyspace(cql3::query_processor&
         , _sp(sp) {
 }
 
+static future<> add_timeout_columns_if_missing(database& db, ::service::migration_manager& mm) noexcept {
+    static std::string_view timeout_columns[] {
+        "read_timeout", "write_timeout", "range_read_timeout", "counter_write_timeout", "truncate_timeout", "cas_timeout", "other_timeout"
+    };
+    try {
+        auto schema = db.find_schema(system_distributed_keyspace::NAME, system_distributed_keyspace::SERVICE_LEVELS);
+        schema_builder b(schema);
+        bool updated = false;
+        for (const std::string_view& col_name : timeout_columns) {
+            bytes options_name = to_bytes(col_name.data());
+            if (schema->get_column_definition(options_name)) {
+                continue;
+            }
+            updated = true;
+            b.with_column(options_name, duration_type, column_kind::regular_column);
+        }
+        if (!updated) {
+            return make_ready_future<>();
```

Comment:
I feel like this should be much simpler, like `co_await execute_cql("ALTER TABLE ... ADD COLUMN ...");`

Reply:
(it can't be done for the bootstrap tables used for schemas, but for these higher level tables, I don't see why we need to C++ it)

Reply:
Original discussion:
https://github.com/scylladb/scylla/pull/7617#discussion_r533335847
https://github.com/scylladb/scylla/pull/7617#issuecomment-736792433
https://github.com/scylladb/scylla/pull/7617#issuecomment-737312128

tldr: we want a well-defined timestamp to avoid schema mismatches. Hence the previous patch which allows custom timestamps, and the C++ below to explicitly update with a well-defined timestamp (0).

Reply:
Aha. Perhaps we could have a clause for DDL to support this (but better to wait for Raft schemas).